### ISSUE TITLE
Multiple quiet login requests (different scopes) from an extension to a single AuthenticationProvider are indistinguishable on the Accounts menu

### ISSUE THREAD

**Issue by gjsjohnmurray** [microsoft/vscode#137614]
![image](https://user-images.githubusercontent.com/6726799/142768457-1f2bd4c1-0137-40cf-b66e-98e11ee86416.png)

> The two quiet logins are now accessible, but the screenshot reveals another issue, which is that the menu item prompts for the two different-scopes requests are identical. I will open a separate issue for this.

_Originally posted by @gjsjohnmurray in https://github.com/microsoft/vscode/issues/137613#issuecomment-974840018_

**Comment by gjsjohnmurray** [comment]
Perhaps `AuthenticationGetSessionOptions` could have a new `hint?: string` property which when passed would replace the cryptic digit 1 in parens at the end of the item text.

Arguably such a hint property could also be used to provide the additional detail text which a non-quiet forceNewSession reauthentication can now achieve by passing `forceNewSession: "Why we need a new session"`. As implemented for #129064 (about to finalize) the extra text can only be added to the **re**-authenticating variant of the modal dialog. The `hint` property I am suggesting could also work for the authenticating variant and for the quiet login menu items.

**Comment by TylerLeonhardt** [comment]
I think we need this first?

https://github.com/microsoft/vscode/issues/124643

**Comment by gjsjohnmurray** [comment]
I don't think that's a dependency, but I agree that until we have it a situation with multiple quiet logins from the same extension to the same authprovider will misbehave.

**Comment by TylerLeonhardt** [comment]
True. And yes thanks for opening this one. Something I've wanted to do is something like one of these pages:
![image](https://user-images.githubusercontent.com/2644648/142950957-6776881d-3dc9-4f7f-b167-c8208bfc0acc.png)

Perhaps what we can do is:
* for `createIfNone: true` we can include the scopes in the modal that opens up
* for `createIfNone: false` we can have clicking on the "Sign in" open a modal that displays the scopes as well (essentially showing the `createIfNone: true` modal

That way:
1. the experience is consistant
2. we're very clear to the user what they are being requested for

Later on it would be nice to allow an Auth Provider to provide human readable names for scopes... so for example in the Microsoft Authentication extension, you could ask for `Task.ReadWrite` but really this is the `Microsoft To-Do (read & write)` permission. This will help the users understand scopes that don't make sense.

So I like:

* add a `hint` just like `forceNewSession`
* display scopes being requested to the user
  * have a mechanism for making them user-readable

**Comment by parithon** [comment]
Any updates on this one? I have a use case which would benefit from the proposed additional features.

**Comment by TylerLeonhardt** [comment]
Just realized this is covered with other issues now.

> add a hint just like forceNewSession

that's https://github.com/microsoft/vscode/issues/117358

> display scopes being requested to the user

https://github.com/microsoft/vscode/issues/96609

however this issue has more details so I will dupe https://github.com/microsoft/vscode/issues/96609 to this.
